Neos.ContentRepositoryRegistry/Classes/Service/CatchUpDeduplicationQueue.php

```diff
@@ -0,0 +1,171 @@
+<?php
+namespace Neos\ContentRepositoryRegistry\Service;
+
+use Neos\Cache\Frontend\FrontendInterface;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\Projection\ProjectionCatchUpTriggerInterface;
+use Neos\ContentRepository\Core\Projection\Projections;
+
+/**
+ *
+ */
+final readonly class CatchUpDeduplicationQueue
+{
+    public function __construct(
+        private ContentRepositoryId $contentRepositoryId,
+        private FrontendInterface $catchUpLock,
+        private ProjectionCatchUpTriggerInterface $projectionCatchUpTrigger
+    ) {}
+
+    public function requestCatchUp(Projections $projections): void
+    {
+        $queuedProjections = $this->triggerCatchUpAndReturnQueued($projections);
+        $attempts = 0;
+        /** @phpstan-ignore-next-line */
```

Comment:
Just out of curiosity: What was PHPStan complaining about?

Reply:
the new `Projections::empty`is in an internal class  if you have suggestions I am happy, we could make Projections not internal or I could not use it, but it seems very convenient.